bin/archive-conf

```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/python -b
+#!/usr/bin/env -S python -b
```

Comment:
I'm pretty sure `env -S` is not very portable.

Reply:
If you expect portage to only run on systems where python is at a fixed location then the further assumption that `env` supports `-S` is inconsequential. Overall this should decrease the amount of work to support POSIX-compliant systems that are not Gentoo.

I spent some time trying to determine how portable this was. I do see manual pages that do not show `-S`, and sometimes invoking `env` with additional arguments behaves as if `-S` was passed. But keep in mind there are already portability problems, e.g. a system may lack `xz`.

There is the additonal concern of what the most common flag, `-b`, accomplishes; to me it seems more appropriate to leave this to a linter but I have no full replacement ready.

Reply:
Critical scripts for the system would better have a fixed path. Maybe one reason is for security, i.e., using env may lead to arbitrary executable to be run. I know Fedora does the same for python scripts.

Reply:
I understand your concern but if you can't trust `env` to be secure then it seems like you also can't trust `python` to be secure. The solution would be to use verity like Android does.

Reply:
You can trust /usr/bin/env, but you may not trust your environment. For example, I have a evil executable named `python` in $PATH that will run `rm -rf /something_important` instead of being a python interpreter. Then when I call emerge, bad things will happen. By using the fixed path, we can be sure /usr/bin/python will always be called and we can trust it because making changing to /usr/bin/python requires root privilege. That is, our thread model is that the attacker won't have root privilege. This is a very common thread model.

Reply:
No, this is security by obscurity. You kill valid use cases because something bad might happen. In the end, bad things happen because of silly tinfoil assumptions hardcoded somewhere.

Reply:
Quote from https://blogs.gnome.org/mcatanzaro/2018/02/16/on-python-shebangs/.

The first option seems to work in all environments, but it is banned in popular distros like Fedora (and I believe also Debian, but I can’t find a reference for this). Using env in shebangs is dangerous because it can result in system packages using non-system versions of python. python is used in so many places throughout modern systems, it’s not hard to see how using #!/usr/bin/env in an important package could badly bork users’ operating systems if they install a custom version of python in /usr/local. Don’t do this.